libenkf/src/enkf_state.c

```diff
@@ -706,15 +706,21 @@ static void enkf_state_internalize_GEN_DATA(enkf_state_type * enkf_state ,
 
 
 
-
 static forward_load_context_type * enkf_state_alloc_load_context(const enkf_state_type * state,
                                                                  run_arg_type * run_arg,
                                                                  stringlist_type * messages) {
-  bool load_summary = ensemble_config_has_impl_type(state->ensemble_config,
-                                                    SUMMARY);
-  if (!load_summary) {
+  bool load_summary = false;
+
+  if (ensemble_config_has_GEN_DATA(state->ensemble_config))
+    load_summary = true;
+
+  if (ensemble_config_has_impl_type(state->ensemble_config, SUMMARY))
+    load_summary = true;
+
+  {
```

Comment:
Please remove these guards surrounding and immediately succeeding `if`s.  They are very confusing (and even more unnecessary).

Reply:
Well - you think they are confusing; I think they improve readability; these are subjective matters.